pulsar-client/src/main/java/org/apache/pulsar/client/impl/schema/LongSchema.java

```diff
@@ -70,9 +73,9 @@ public Long decode(byte[] bytes) {
         }
         validate(bytes);
         long value = 0L;
-        for (byte b : bytes) {
+        for (int i = 0; i < 8; i++) {
```

Comment:
I understand the change here is for the optimization you have in presto. but it should be done in a clearer way, you should introduce a new decode method `decode(byte[] data, int offset, int len)`.